src/back/deps/svetit/shared/db/utils/idstuple.hpp

```diff
@@ -26,9 +26,33 @@ struct FilterIdsTuple {
 };
 
 template<typename T>
-struct FilterIdsTuple<T, typename enable_if_type<typename T::FilterIds>::type> {
+struct FilterIdsTuple<T, typename svetit::utils::enable_if_type<typename T::FilterIds>::type> {
 	using type = T::FilterIds;
 	static constexpr auto Get() { return T::FilterIds::Get(); }
 };
 
+template<typename T, class Enable = void>
+struct NoInsertIdsTuple {
+	using type = std::index_sequence<>;
+	static constexpr std::tuple<> Get() { return {}; }
+};
```

Comment:
По умолчанию должна быть первая колонка указана.
```suggestion
struct NoInsertIdsTuple {
	using type = std::index_sequence<0>;
	static constexpr std::tuple<std::size_t> Get() { return {0}; }
};
```

или может быть даже так:
```suggestion
struct NoInsertIdsTuple : IdsTuple<T, Enable> {};
```

Reply:
Если будет первая колонка указана, то NoInsertIds всегда будет с первым (нулевым) элементом даже, если мы этого не указываем в описании структуры типа. В результате в связующих таблицах, где NoInsertIds должен быть пустым, будет ошибка. Поскольку будем пытаться не вставлять в один из столбцов при insert (в первый), который не может быть пустым по структуре таблицы.

Reply:
В связующих таблицах нужно добавить NoInsertIds пустой.

Reply:
> В связующих таблицах нужно добавить NoInsertIds пустой.

а как это сделать? я пробовал, когда ты был в отпуске, у меня не получилось.

Reply:
А какая ошибка если так:
```c++
using NoInsertIds = db::Ids<>;
```

Reply:
> А какая ошибка если так:
> 
> ```c++
> using NoInsertIds = db::Ids<>;
> ```



> struct NoInsertIdsTuple : IdsTuple<T, Enable> {};

теперь с 
```
struct NoInsertIdsTuple : IdsTuple<T, Enable> {};
```
и при указании 
```
using NoInsertIds = db::Ids<>;
```
ошибки нет. Сейчас везде внедрю.

Reply:
Внедрил, всё же ошибки есть.
Например, вот:
```
3: INFO     [POST /project/value-view] start handling request_body={
3:   "diTypeId": 1,
3:   "id": 0,
3:   "value": "Test Value",
3:   "view": "Test View"
3: }
3: DEBUG    [POST /project/value-view] Result checked
3: DEBUG    [POST /project/value-view] Prepare successfully sent
3: DEBUG    [POST /project/value-view] Result checked
3: ERROR    [POST /project/value-view] Fatal error occurred: ОШИБКА:  значение NULL в столбце "space_id" отношения "value_view" нарушает ограничение NOT NULL\nDETAIL:  Ошибочная строка содержит (3, null, 1, Test Value, Test View).\n
3: DEBUG    [POST /project/value-view] Ready to throw
3: ERROR    [POST /project/value-view] Method: POST Url: /project/value-view User: d2f9924d-a69a-4aec-b9c0-c80171d3ed86 Err: ОШИБКА:  значение NULL в столбце "space_id" отношения "value_view" нарушает ограничение NOT NULL\nDETAIL:  Ошибочная строка содержит (3, null, 1, Test Value, Test View).\n Body: {"id": 0, "diTypeId": 1, "value": "Test Value", "view": "Test View"}
3: ERROR    [POST /project/value-view] Response finished meta_code=500 response_body={
3:   "err": "Internal error"
3: }
3: INFO     [POST /tests/log_flush] start handling request_body={
3:   "logger_name": null
3: }
```

Reply:
Используй первый вариант
```c++
struct NoInsertIdsTuple {
	using type = std::index_sequence<0>;
	static constexpr std::tuple<std::size_t> Get() { return {0}; }
};
```

Второй дал ошибку, потому что поле Ids у нас содержит поля обязательные для методов Get, Update и Delete, но которые по факту не pk. Поэтому когда мы наследовались от  IdsTuple мы получили space_id в списке Idшников которые не нужно вставлять. В будущем наверное переименуем эти структуры так чтобы было более явно их назначение.